src/libServer/LookupServer.cpp

```diff
@@ -913,7 +912,8 @@ string LookupServer::GetEthCall(const Json::Value& _json) {
     }
     if (_json.isMember("gasLimit")) {
       const auto& gasLimit_str = _json["gasLimit"].asString();
-      gasRemained = strtoull(gasLimit_str.c_str(), NULL, 0);
+      gasRemained =
+          min(gasRemained, (uint64_t)strtoull(gasLimit_str.c_str(), NULL, 0));
```

Comment:
Why not prefer `std::stol` ?   This one will return `ULONG_MAX` on error, and mess with `errno`.  Because of the `min` it will end up fine, but it is soo non-obvious...